src/Pathfinding/Pather2.cpp

```diff
@@ -31,76 +31,75 @@ void print_queue(std::queue<Pather2::queueNode> q) {
-               // set adj point as visited and push it to points to be checked(q)
-               struct PointXY test = {row, col};
-               std::queue<PointXY> addAdjPointToPath = curr.path;
-               addAdjPointToPath.push(test);
-               Pather2::queueNode adjNode = {{row, col}, curr.dist + 1, addAdjPointToPath};
-               map[(int)adjNode.pt.x][(int)adjNode.pt.y] = true;
-               q.push(adjNode);
-           }
-       }
-      
-   }
-   return error;
+std::queue<PointXY> Pather2::BFS(PointXY dest){
+    int rowNum[] = {-1,0,0,1, -1, 1, 1, -1};
+    int rowCol[] = {0,-1,1,0, -1, -1, 1, 1};
+
+    PointXY src = {5,5};
+    std::queue<PointXY> error;
+    error.push(src);
+    //check if src and dest are represented with 1 not 0
+    if ((obsMap.obstacle_map[(int)src.x][(int)src.y]) || (obsMap.obstacle_map[(int)dest.x][(int)dest.y])){
+        //todo: change destination if destination is blocked
+        return error;
+    }
+
+    obsMap.obstacle_map[(int)src.x][(int)src.y] = true;
+    std::queue<Pather2::queueNode> q;
+    std::queue<PointXY> newQueue;
+    newQueue.push(src);
+    Pather2::queueNode s = {src, 0, newQueue};
+    q.push(s);
+
+    // q is the queue of points that are waiting to be checked
+    // curr.path is the path leading up to that point
+
+    while (!q.empty()){
+        //get front point in points to be checked and make it current
+        Pather2::queueNode curr = q.front();
+        PointXY pt = curr.pt;
+        
+        //check if current point is destination node
+        if ((pt.x == dest.x) && (pt.y == dest.y)){
+            //return first point after source in path to destination
+            std::queue<PointXY> temp = curr.path;
+            temp.pop();
+            std::cout << "Final path: ";
+            print_point(temp);
+            
+            return temp;
+        }
+        
+        // pop current point off point to be checked
+        q.pop();
+        
+        // iterate through adj points
+        for (int i = 0; i < 8; i++){
+            int row = pt.x + rowNum[i];
+            int col = pt.y + rowCol[i];
+            
+            //((row >= 0) && (row < 21) && (col >= 0) && (col < 21))
+            // check if adj point is not out of bounds and is not an obstacle
+            if (((row >= 0) && (row < 21) && (col >= 0) && (col < 21)) && !obsMap.obstacle_map[row][col]){
+                // set adj point as visited and push it to points to be checked(q)
+                struct PointXY test = PointXY{(float)row, (float)col};
+                std::queue<PointXY> addAdjPointToPath = curr.path;
+                addAdjPointToPath.push(test);
+                Pather2::queueNode adjNode = {PointXY{(float)row, (float)col}, curr.dist + 1, addAdjPointToPath};
+                obsMap.obstacle_map[(int)adjNode.pt.x][(int)adjNode.pt.y] = true;
+                q.push(adjNode);
+            }
+        }
+        
+    }
+    return error;
```

Comment:
Oddly, nowhere in this method is `error` ever updated. So this function will always return (5,5). Seems like there are probably a lot of bugs in here that we should take a look at in a follow-up pull request. We should also take a look at PatherTest to figure out why it hasn't caught any of these bugs.